src/script/components/MessagesList/index.tsx

```diff
@@ -179,9 +179,10 @@ const MessagesList: React.FC<MessagesListParams> = ({
     const isLastDeliveredMessage = lastDeliveredMessage?.id === message.id;
 
     const visibleCallback = getVisibleCallback(conversation, message);
+    const key = (message.id || 'message-') + message.timestamp();
```

Comment:
For some messages `id` is undefined, and tine `timestamp()` is ko.observable so it has to be accessed this way not just `timestamp`.

Reply:
It will provide duplicated key, as the system messages have the same string as `id`